xonsh/built_ins.py

```diff
@@ -266,15 +267,22 @@ def _get_runnable_name(fname):
 
 
 def _is_binary(fname, limit=80):
-    with open(fname, 'rb') as f:
-        for i in range(limit):
-            char = f.read(1)
-            if char == b'\0':
-                return True
-            if char == b'\n':
-                return False
-            if char == b'':
-                return False
+    try:
+        with open(fname, 'rb') as f:
+            for i in range(limit):
+                char = f.read(1)
+                if char == b'\0':
+                    return True
+                if char == b'\n':
+                    return False
+                if char == b'':
+                    return False
+    except IOError as exc:
+        if exc.errno != errno.EACCES:
```

Comment:
Thanks for the PR @larsks! I think that this lets too many files through, though.  I think that this will allow the execution of files with any permision.  Is there any way that you could please refactor this to check for the correct permission bits? Thanks again